File: Intermediate-sales-commission-365/Python/UserDto.py
class UserDto: 
    revenue = {}
    expenses = {}
    commission = 0.062

    def __init__(self, name, revenue = {}, expenses = {}):
        self.name = name
        self.revenue = self.revenue.copy()
        self.expenses = self.expenses.copy()

    def add_revenue_item(self, item, value):
        self.revenue[item] = value
      
    def add_expenses_item(self, item, value):
        self.expenses[item] = value

    def get_commission(self):
        overall_profit = 0
        for key in self.revenue:
            overall_profit += max(float(self.revenue[key]) - float(self.expenses[key]), 0.0)
        
        return round(overall_profit * self.commission, 2)

File: Intermediate-sales-commission-365/C#/CommissionCalculatorCSharp/UserDto.cs
using System;
using System.Collections.Generic;

namespace CommissionCalculatorCSharp
{
    public class UserDto
    {
        private double _commission = 0.062;
        public string Name {get;set;}
        public Dictionary<string, double> Revenue {get;set;}
        public Dictionary<string, double> Expenses {get;set;}

        public UserDto(string name)
        {
            Name = name;
            Revenue = new Dictionary<string, double>();
            Expenses = new Dictionary<string, double>();
        }

        public double getCommission() 
        {
            double overallProfit = 0.0;

            foreach(var item in Revenue)
            {
                overallProfit += Math.Max(Revenue[item.Key] - Expenses[item.Key], 0);
            }

            return Math.Round(overallProfit * _commission, 2);
        }
    }
}